src/services/completions.ts

```diff
@@ -429,7 +435,12 @@ namespace ts.Completions {
     }
 
     function getSourceFromOrigin(origin: SymbolOriginInfo | undefined): string | undefined {
-        return origin && originIsExport(origin) ? stripQuotes(origin.moduleSymbol.name) : undefined;
+        if (originIsExport(origin)) {
+            return stripQuotes(origin.moduleSymbol.name);
+        }
+        if (origin?.kind === SymbolOriginInfoKind.ThisType) {
+            return "" + SymbolOriginInfoKind.ThisType;
+        }

```

Comment:
@uniqueiniquity Might have thoughts too.

Reply:
All good in VS - we never display the source property on `CompletionEntry`, but we do display the source property on `CompletionEntryDetails`, as allowed by the comment there.

Reply:
I am not sure how vscode uses source property..  @mjbvz might know but I don't think we should use `SymbolOriginaInfoKind` for that since its a enum with number values.

Reply:
VS Code also renders the `source` in the suggestion details: https://github.com/microsoft/vscode/blob/9d25faad7fbf3255ca32959be47466107555cb08/extensions/typescript-language-features/src/features/completions.ts#L715

Reply:
That `source` is from the completion details response, which is calculated separately and intended to be human-readable. It will remain blank for `this.` completion details. Looks like `source` on the `CompletionEntry` itself is never displayed. I’ve replaced the `SymbolOriginInfoKind` stringification with a dedicated string enum per @sheetalkamat’s feedback.